.github/workflows/hq-smoke-tests.yml

```diff
@@ -142,10 +143,28 @@ jobs:
           attachments: ${{ github.workspace }}/report_${{ matrix.environment }}.html
           in_reply_to: ${{ fromJSON(steps.configure_email.outputs.result).reference }}
 
+      - name: Set slack vars
+        if: ${{ always() }}
+        id: configure_slack
+        uses: actions/github-script@v6
+        env:
+            JOB_STATUS: ${{ job.status }}
+            CC_ENV: ${{ matrix.environment }}
+        with:
+          result-encoding: string
+          script: |
+            const {CC_ENV, JOB_STATUS} = process.env
+            let SLACK_WEBHOOK_URL = '${{ secrets.SLACK_WEBHOOK_URL_SMOKE }}'
+            if (context.event_name == 'repository_dispatch' && CC_ENV == 'production') {
+              SLACK_WEBHOOK_URL = '${{ secrets.SLACK_WEBHOOK_URL_PROD_RESULTS }}'
+            }
+            
+            return SLACK_WEBHOOK_URL
+
       - name: Post to Slack channel on Failure
         id: slack
         uses: slackapi/slack-github-action@v1.23.0
-        if: failure()
+        if: ${{ success() || failure() }}
```

Comment:
```suggestion
         if: ${{ matrix.environment == 'production' || failure() }}
```

Reply:
I tried this change but it didn't work as expected.